src/pages/ReimbursementAccount/BankAccountPlaidStep.js

```diff
@@ -39,109 +38,102 @@ const defaultProps = {
     plaidLinkOAuthToken: '',
 };
 
-class BankAccountPlaidStep extends React.Component {
-    constructor(props) {
-        super(props);
-        this.validate = this.validate.bind(this);
-        this.submit = this.submit.bind(this);
-    }
+function BankAccountPlaidStep(props) {
+    const {plaidData, receivedRedirectURI, plaidLinkOAuthToken} = props;
+    const {translate} = useLocalize();
 
-    validate(values) {
+    const validate = useCallback((values) => {
         const errorFields = {};
         if (!values.acceptTerms) {
             errorFields.acceptTerms = 'common.error.acceptTerms';
         }
 
         return errorFields;
-    }
+    }, []);
 
-    submit() {
-        const selectedPlaidBankAccount = _.findWhere(lodashGet(this.props.plaidData, 'bankAccounts', []), {
-            plaidAccountID: lodashGet(this.props.reimbursementAccountDraft, 'plaidAccountID', ''),
+    const submit = useCallback(() => {
+        const selectedPlaidBankAccount = _.findWhere(lodashGet(plaidData, 'bankAccounts', []), {
+            plaidAccountID: lodashGet(props.reimbursementAccountDraft, 'plaidAccountID', ''),
         });
 
         const bankAccountData = {
             routingNumber: selectedPlaidBankAccount.routingNumber,
             accountNumber: selectedPlaidBankAccount.accountNumber,
             plaidMask: selectedPlaidBankAccount.mask,
             isSavings: selectedPlaidBankAccount.isSavings,
-            bankName: lodashGet(this.props.plaidData, 'bankName') || '',
+            bankName: lodashGet(plaidData, 'bankName') || '',
             plaidAccountID: selectedPlaidBankAccount.plaidAccountID,
-            plaidAccessToken: lodashGet(this.props.plaidData, 'plaidAccessToken') || '',
+            plaidAccessToken: lodashGet(plaidData, 'plaidAccessToken') || '',
         };
         ReimbursementAccount.updateReimbursementAccountDraft(bankAccountData);
 
-        const bankAccountID = lodashGet(this.props.reimbursementAccount, 'achData.bankAccountID') || 0;
+        const bankAccountID = lodashGet(props.reimbursementAccount, 'achData.bankAccountID') || 0;
         BankAccounts.connectBankAccountWithPlaid(bankAccountID, bankAccountData);
-    }
+    }, [props.reimbursementAccount, props.reimbursementAccountDraft, plaidData]);
 
```

Comment:
NAB Why  `reimbursementAccount` or  `reimbursementAccountDraft` is not destructured?

Reply:
good catch! updated to de-structure other props as well